### DISCUSSION TITLE
Is it possible open url outside of Firefox to totally separate hierarchy?

### DISCUSSION THREAD

**Discussion by mjo** [piroor/treestyletab#3313] [Q&A] [answered]
Hi!

If I have following hierarchy ongoing:

<pre><code>
My physics studies(1)
     Higgs boson(2)
    Galileo Galilei (3)
    .... (4)
My Favorite bands (5)
    Band A (6)
    Band B (7)
.... (8)
.... (9)
</code></pre>

And I'm currently in 3rd tab,  so Galileo, and open url e.g. from Outlook, it will open that link to location 4.

Can I somehow configure that all those external urls will open to location 8, next one to 9 and so forth?

Motivation: you have separate topics/ research ongoing and when you will open something new from outside, it won't pollute existing hierarchy but keep it totally separate thing

BR

**Comment by irvinm** [comment, marked as answer]
If you have about:config option "browser.tabs.insertAfterCurrent" = true, and if you were on tab (3), then any tab you created (either from Firefox or externally would be create next to that one.

**Reply by irvinm** [reply]
Another choice would be to place all "external tabs" into a different tree with this addon:  https://addons.mozilla.org/en-US/firefox/addon/tst-auto-group-tabs/

**Reply by mjo** [reply]
> If you have about:config option "browser.tabs.insertAfterCurrent" = true, and if you were on tab (3), then any tab you created (either from Firefox or externally would be create next to that one.

Thank you very much, it works greatly now 👍

**Comment by mjo** [comment]
Above has been working greatly. How about opening those "external tabs" to the top of the tree? So against above sorting behavior

**Reply by irvinm** [reply]
My suggestion on this one is to ask for an enhancement to https://addons.mozilla.org/en-US/firefox/addon/tst-auto-group-tabs/ to allow external tabs to simply be inserted to the top of the tree.  I took a quick look at the code and it seems fair easy to add an option for "regular grouping" vs. "insert at top".  (Well, easy for @piroor ... :)

I think in this block of code, there can just be a branch to insert in a group (attachTabsToGroup) vs. "insert at top".

https://github.com/piroor/tst-auto-group-tabs/blob/0f34c27c6b5183ea68fec3d599f5c34b5c612dc2/background/background.js#L175-L193

The other option is to create another addon by pulling the logic defining an "external tab" (monitoring if focus was set to the browser or "outside the browser" when a new tab is created) and then if deemed "external" to just move the tab to the top of the tree.
